### PR TITLE
[java] TooManyMethods - improve getter/setter detection (#3729)

### LINKED ISSUE #3729: [java] TooManyMethods ignores "real" methods which are named like getters or setters
**Affects PMD Version:** any

**Rule:** [TooManyMethods](https://pmd.github.io/latest/pmd_rules_java_design.html#toomanymethods)

**Description:**
This false negative is from a source code comment.
Currently the rule ignores methods, that start with "get", "set", or "is" - which are considered simple getter/setter methods. But if these methods do more complex things (despite they are named like setters, e.g. if the setter contains validating logic), then this should maybe count towards the total method count...

Original comment:

```
                          FIXME: Refine XPath to discard 'get' and 'set' methods with Block no more than 3 lines,
                                something like this:
                                    not (
                                            (
                                                starts-with(@Name,'get')
                                                or
                                                starts-with(@Name,'set')
                                                or
                                                starts-with(@Name,'is')
                                            )
                                            and (
                                                    (
                                                        (../Block/attribute::endLine)
                                                         -
                                                        (../Block/attribute::beginLine)
                                                    ) <= 3
                                            )
                                        )
                                This will avoid discarding 'real' methods...
```

### PR DESCRIPTION
**Pull request by 341816041**
## Describe the PR

Add a condition for judging a method as a getter setter

## Related issues

- Fixes #3729 

## Ready?

- [1] Added unit tests for fixed bug/feature
- [0] Passing all unit tests
- [0] Complete build `./mvnw clean verify` passes (checked automatically by github actions)(it goes wrong at disk test)
- [0] Added (in-code) documentation (if needed)

### CODE REVIEW

File: pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/design/xml/TooManyMethods.xml [hunk 1]
```diff
@@ -110,6 +110,40 @@ public class OuterClass {
         @Override
         public void method7() {}
     }
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>#3729 TooManyMethods ignores "real" methods which are named like getters or setters</description>
+        <expected-problems>0</expected-problems>
```
Comment (adangel) [inline pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/design/xml/TooManyMethods.xml:119]: Shouldn't we expect here a violation?

The class has 22 methods, which are `> 10`... It looks like the trick with `count(../Block/BlockStatement)<=1)` didn't work out?

**Comment by C-Otto** [comment]
I just noticed that the documentation in https://pmd.github.io/latest/pmd_rules_plsql_design.html still includes the exclusion of get/set/is methods. Could you please update it, too?

**Comment by adangel** [comment]
@C-Otto The file you mention is about *PLSQL* - the java version is updated: https://pmd.github.io/latest/pmd_rules_java_design.html#toomanymethods

> still includes the exclusion of get/set/is methods

That's the whole point - exclude getter/setter/methods from the TooManyMethods rule, but keep e.g. getters, that do more (that's the `count(Block/BlockStatement) <= 1` condition).

Please create a new issue, if you think, something is wrong. Thanks.

**Comment by C-Otto** [comment]
Everything is fine, sorry for the confusion!

**Pull request merged**